packages/shared/lib/core/wallet/actions/getClient.ts

```diff
@@ -2,7 +2,11 @@ import { Client } from '@iota/sdk/out/client'
 import { selectedWalletId } from '@core/wallet/stores'
 import { api } from '@core/api'
 import { get } from 'svelte/store'
+import { getWalletById } from '../../profile'
 
 export function getClient(walletId: string = get(selectedWalletId)): Promise<Client> {
-    return api.getClientFromWallet(walletId)
+    const wallet = getWalletById(walletId)
```

Comment:
just curious, but i don't understand why we need to check `wallet` :thinking:  :pray:

Reply:
Just to prevent any race condition between the destruction of the wallet and any background polling that we have on firefly, it's just in case.